crates/wasm-mutate/src/info.rs

```diff
@@ -309,6 +309,31 @@ impl<'a> ModuleInfo<'a> {
         module
     }
 
+    /// Move a section from index `src_idx` to `dest_idx` in the Wasm module
+    pub fn move_section(&self, src_idx: usize, dest_idx: usize) -> wasm_encoder::Module {
+        log::trace!(
+            "moving section from index {} to index {}",
+            src_idx,
+            dest_idx
+        );
+        assert!(src_idx < self.raw_sections.len());
+        assert!(dest_idx < self.raw_sections.len());
+        assert_ne!(src_idx, dest_idx);
+        let mut module = wasm_encoder::Module::new();
+        self.raw_sections.iter().enumerate().for_each(|(i, s)| {
+            if src_idx < dest_idx && i == dest_idx {
+                module.section(&self.raw_sections[src_idx]);
+            }
+            if i != src_idx {
+                module.section(s);
+            }
+            if dest_idx < src_idx && i == dest_idx {
+                module.section(&self.raw_sections[src_idx]);
+            }
```

Comment:
I paused for a second here to try to review this carefully to make sure something wasn't encoded twice since the `if` statements all execute on each iteration of the loop. This looks good, but would this simplification also work perhaps?

```rust
let refs = self.raw_sections.iter().collect::<Vec<_>>();
refs.swap(src_idx, dest_idx);
let mut module = wasm_encoder::Module::new();
for section in refs {
    module.section(section);
}
module
```

Reply:
That would also reorder the sections, but it would do so slightly differently. Rather than moving `src` to `dest` and keeping the order of all the others the same, it would also affect the placement of where the section at dest_idx is if I'm understanding correctly. So rather than a reorder like this:
```
// [a, b, c]
move_section(0, 2)
// [b, c, a]
```
it would reorder like this:
```
// [a, b, c]
move_section(0, 2)
// [c, b, a]
```

I think that's probably fine? I'll defer to you and @fitzgen since y'all have a bit more context here than I do.

Reply:
Oh true yeah, could `.remove` and `.insert` on vectors be used for that then?

Reply:
I think that would work, but at this point I don't think it is any nicer than what is currently in the PR. But I don't feel super strongly.

Reply:
True yeah, in that case I'll merge!